hardware/raspberrypi/bootmodes/bootflow_2711.md

```diff
@@ -1,31 +1,74 @@
-# Pi4 Bootflow
+# Raspberry Pi4, Pi400 and CM4 bootflow
 
-The Pi4 with the BCM2711 SoC has a new, more sophisticated boot process. The addition of an EEPROM means that the `bootcode.bin` file found in `/boot` is no longer required. Details of the EEPROM can be found [here](../booteeprom.md).
+This page describes the bootflow for BCM2711 based products. The main difference betweeen this and previous products is that the second stage bootloader is loaded from an SPI flash [EEPROM](../booteeprom.md) on BCM2711 instead of the bootcode.bin file on previous chips.
 
-The boot flow for the Pi4 is as follows:
+The bootflow for the ROM (first stage) is as follows:-
 
 * BCM2711 SoC powers up
-* On board bootrom checks for bootloader recovery file (recovery.bin) on the SD card. If found, it executes it to flash the EEPROM and recovery.bin triggers a reset.
-* Otherwise, the bootrom loads the main bootloader from the EEPROM.
-* Bootloader checks it's inbuilt BOOT_ORDER configuration item to determine what type of boot to do.
-  * SD Card
-  * Network
-  * USB mass storage
+* Read OTP to determine if the `nRPIBOOT` GPIO is configured
+* If nRPIBOOT GPIO is high or OTP does NOT define `nRPIBOOT` GPIO 
+   * Check OTP to see if recovery.bin can be loaded from SD/EMMC
+      * If SD recovery.bin is enabled then check primary SD/EMMC for `recovery.bin`
+         * Success - run `recovery.bin`
+         * Fail - continue
+   * Check SPI EEPROM for second stage loader
+      * Success - run second stage bootloader 
+      * Fail - continue
+* While True
+   * Attempt to load recovery.bin from [USB device boot](../../computemodule/cm-emmc-flashing.md)
+      * Success - run `recovery.bin`
+      * Fail - retry USB device boot
 
+N.B. Currently only CM4 reserves a GPIO for `nRPIBOOT`
 
-## SD Card Boot
-The bootloader loads the files in the [boot folder](../../../configuration/boot_folder.md) according to the [boot options](../../../configuration/config-txt/boot.md) in config.txt
+## recovery.bin
+`recovery.bin` is a minimal second stage program used to reflash the bootloader SPI EEPROM image.
```

Comment:
This paragraph feels a bit sparse - could maybe add " which is explained in more detail [here](../booteeprom.md)." ? The bits about `pieeprom.upd` will probably be relevant to anybody reading this bootflow page :)

Reply:
Made the text smaller and linked out to booteeprom.md

Reply:
In the latest commit you've edited the recovery.bin part at the bottom of the page, but the bit I was referring to here is the recovery.bin part mid-way through the page.